compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -4728,27 +4728,30 @@ public BType getXMLSequenceType(BType xmlSubType) {
 
     public void visit(BLangXMLSequenceLiteral bLangXMLSequenceLiteral) {
         BType expType = types.getReferredType(this.expType);
-        if (expType.tag != TypeTags.XML && expType.tag != TypeTags.UNION && expType.tag != TypeTags.XML_TEXT
-        && expType != symTable.noType) {
-            dlog.error(bLangXMLSequenceLiteral.pos, DiagnosticErrorCode.INCOMPATIBLE_TYPES, this.expType,
-                    "XML Sequence");
-            resultType = symTable.semanticError;
-            return;
-        }
-
-        List<BType> xmlTypesInSequence = new ArrayList<>();
+        Set<BType> xmlTypesInSequence = new LinkedHashSet<>();
 
         for (BLangExpression expressionItem : bLangXMLSequenceLiteral.xmlItems) {
             resultType = checkExpr(expressionItem, env, this.expType);
```

Comment:
Is this line correct? If the expected type is `xml<T>`, shouldn't we use `T` here for each individual member?

For example, for

```ballerina
public function main() {
    xml<xml:Comment> _ = xml `<!--a-->foo<!--b-->`;
}
```

the error is currently

```
ERROR [foo.bal:(2:39,2:42)] incompatible types: expected 'xml<xml:Comment>', found 'xml:Text'
```

Shouldn't it be

```
ERROR [foo.bal:(2:39,2:42)] incompatible types: expected 'xml:Comment', found 'xml:Text'
```